src/components/filters/DateRangeFilter.tsx

```diff
@@ -0,0 +1,174 @@
+'use client'
+
+import { DatePickerField } from '@/components/DatePicker'
+import { Button } from '@/components/ui/button'
+import { QuickDateFilter } from '@/constants/quickDateFilters'
+import { ChevronDown } from 'lucide-react'
+import { parseAsString, useQueryStates } from 'nuqs'
+import { useCallback, useEffect, useRef, useState } from 'react'
+
+type DateRangeFilterProps = {
+  startDate: string
+  endDate: string
+  quickFilters: QuickDateFilter[]
+  title?: string
+  defaultOpen?: boolean
+  showBottomBorder?: boolean
+}
+
+export const DateRangeFilter = ({
+  startDate: initialStartDate,
+  endDate: initialEndDate,
+  quickFilters,
+  title = 'Date Range',
+  defaultOpen = true,
+  showBottomBorder = true,
+}: DateRangeFilterProps) => {
+  const [{ startDate, endDate }, setDateParams] = useQueryStates(
+    {
+      startDate: parseAsString.withDefault(''),
+      endDate: parseAsString.withDefault(''),
+    },
+    {
+      history: 'push',
+      shallow: false,
+    },
+  )
+
+  const [filterType, setFilterType] = useState('')
```

Comment:
⛏️ I know I wrote this, but `filterType` doesn't make sense to me. Can we change it to `selectedFilter`, or something else?